airbyte-integrations/connectors-performance/destination-harness/src/main/java/io/airbyte/integrations/destination_performance/PerformanceTest.java

```diff
@@ -0,0 +1,238 @@
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+@Slf4j
+public class PerformanceTest {
+
+  public static final int PORT1 = 9877;
+  public static final int PORT2 = 9878;
+  public static final int PORT3 = 9879;
+  public static final int PORT4 = 9880;
+
+  public static final Set<Integer> PORTS = Set.of(PORT1, PORT2, PORT3, PORT4);
+
+  public static final double MEGABYTE = Math.pow(1024, 2);
+  private final String imageName;
+  private final JsonNode config;
+  private final ConfiguredAirbyteCatalog catalog;
+
+  private DefaultAirbyteDestination destination;
+
+  PerformanceTest(final String imageName, final String config, final String catalog) throws JsonProcessingException {
+    final ObjectMapper mapper = new ObjectMapper();
+    this.imageName = imageName;
+    this.config = mapper.readTree(config);
+    this.catalog = Jsons.deserialize(catalog, ConfiguredAirbyteCatalog.class);
+  }
+
+  void runTest() throws Exception {
+    KubePortManagerSingleton.init(PORTS);
+
+    final KubernetesClient fabricClient = new DefaultKubernetesClient();
+    final String localIp = InetAddress.getLocalHost().getHostAddress();
+    final String kubeHeartbeatUrl = localIp + ":" + 9000;
+    final var workerConfigs = new WorkerConfigs(new EnvConfigs());
+    final var processFactory = new KubeProcessFactory(workerConfigs, "default", fabricClient, kubeHeartbeatUrl, false);
+    final ResourceRequirements resourceReqs = new ResourceRequirements()
+        .withCpuLimit("2.5")
+        .withCpuRequest("2.5")
+        .withMemoryLimit("2Gi")
+        .withMemoryRequest("2Gi");
+    final var allowedHosts = new AllowedHosts().withHosts(List.of("*"));
+    final var jobRoot = "/";
+
+    final var dstIntegtationLauncher = new AirbyteIntegrationLauncher(
+        "1",
+        0,
+        this.imageName,
+        processFactory,
+        resourceReqs,
+        allowedHosts,
+        false,
+        new EnvVariableFeatureFlags());
+    this.destination = new DefaultAirbyteDestination(dstIntegtationLauncher);
+    final WorkerDestinationConfig dstConfig = new WorkerDestinationConfig()
+        .withDestinationConnectionConfiguration(this.config)
+        .withState(null)
+        .withCatalog(convertProtocolObject(this.catalog, io.airbyte.protocol.models.ConfiguredAirbyteCatalog.class));
+
+    log.info("reader first line");
+    BufferedReader reader = new BufferedReader(new InputStreamReader(
+        new URL("https://storage.googleapis.com/airbyte-performance-testing-public/sample-data/faker_1m/users.csv").openStream(),
+        StandardCharsets.UTF_8));
+    final var columnsString = reader.readLine();
+    log.info("*** columns string: {}", columnsString);
+    final Pattern pattern = Pattern.compile(",");
+    final var columns = Arrays.asList(pattern.split(columnsString));
+    log.info("*** columns {}", columns);
+
+    log.info("Destination starting");
+    destination.start(dstConfig, Path.of(jobRoot));
```

Comment:
NOTES: 
- Valid config is passed
- Hangs after calling into this method (this follows from the same call platform makes https://github.com/airbytehq/airbyte-platform/blob/424530343a611a26b6eed3a984962e5b71dd72ef/airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java#L201
- Rebased on top of `master` since Snowflake has had some major changes in the past week

NEXT STEPS:
- Apply this performance harness against Destination BigQuery to see if this is a Snowflake related issue